project/src/components/user/user.tsx

```diff
@@ -1,22 +1,51 @@
 import {Link} from 'react-router-dom';
+import {useAppDispatch, useAppSelector} from '../../hooks';
+import { logoutAction } from '../../store/api-action';
+import {AppRoute, AuthorizationStatus} from '../../const';
+
 
 export default function User(): JSX.Element {
+  const dispatch = useAppDispatch();
+  const userData = useAppSelector((state) => state.userData);
+  const authorizationStatus = useAppSelector((state) => state.authorizationStatus);
+
   return (
     <nav className="header__nav">
       <ul className="header__nav-list">
-        <li className="header__nav-item user">
-          <Link className="header__nav-link header__nav-link--profile" to="#">
-            <div className="header__avatar-wrapper user__avatar-wrapper">
-            </div>
-            <span className="header__user-name user__name">Oliver.conner@gmail.com</span>
-            <span className="header__favorite-count">3</span>
-          </Link>
-        </li>
-        <li className="header__nav-item">
-          <Link className="header__nav-link" to="/login">
-            <span className="header__signout">Sign out</span>
-          </Link>
-        </li>
+        {(userData && authorizationStatus === AuthorizationStatus.Auth) ?
+          <>
+            <li className="header__nav-item user">
+              <Link className="header__nav-link header__nav-link--profile" to= {AppRoute.Favorites}>
+                <div
+                  className="header__avatar-wrapper user__avatar-wrapper"
+                  style= {{backgroundImage: `url(${userData?.avatarUrl})`}}
```

Comment:
```suggestion
                  style= {{backgroundImage: `url(${userData.avatarUrl})`}}
```
Выше ведь есть проверка что `userData` не пустой, поэтому здесь к нему можно обращаться без проверки.